internal_packages/keybase/lib/keybase-user.cjsx

```diff
@@ -0,0 +1,237 @@
+
+    key = _.find(keys, (key) =>
+      # not sure if the toString is necessary?
+      return key.key? and key.key.get_pgp_fingerprint().toString('hex') == targetKey.key.get_pgp_fingerprint().toString('hex')
+    )
+
+    if key == undefined
+      return null
+    else
+      return key
+
+  _importKey: =>
+    # opens a popover requesting user to enter 1+ emails to associate with a
+    # key - a button in the popover then calls _save to actually import the key
+    #popoverTarget = ReactDOM.findDOMNode(@refs.button).getBoundingClientRect()
+    popoverTarget = ReactDOM.findDOMNode(@refs.button).getBoundingClientRect()
+
+    Actions.openPopover(
+      <EmailPopover onPopoverDone={ @_popoverDone } />,
+      {originRect: popoverTarget, direction: 'left'}
+    )
+
+  _popoverDone: (addresses) =>
+    # closes the popover, saves a key if an email was entered
+    {profile} = @props
+    keybaseUsername = profile.keybase_user.components.username.val
+
+    if addresses.length < 1
+      # no email addresses added, nop
+      return
+    else
+      @_save(keybaseUsername, addresses[0])
+
+    if addresses.length > 1
+      # add any extra ddresses the user entered
+      _.each(addresses.slice(1), (address) =>
+        @_addEmail(address)
+      )
+
+  _save: (keybaseUsername, address) =>
+    # save/import a key from keybase
+    kb.getKey(keybaseUsername, (key) =>
+      if not key?
+        console.error "Unable to fetch key for #{keybaseUsername}"
+      else
+        PGPKeyStore.saveNewKey(address, key, true) # isPub = true
+    )
+    return
+
+  _delete: (event) =>
+    # delete a locally saved key
+    email = event.target.attributes['data-email'].value
+    keys = PGPKeyStore.pubKeys(email)
+    if keys? and keys.length == 1
+      key = keys[0]
+    else if keys.length > 1
+      # TODO key metadata objects need a unique ID
+      key = @_matchKeys(@props.profile.key, keys)
+      #console.error "Multiple keys associated with #{email}, unable to determine which to remove from"
+
+    if key?
+      PGPKeyStore.deleteKey(key)
+    else
+      console.error "Unable to fetch key for #{email}"
+
+  _addEmail: (email) =>
+    # associate another email address with this key
+    PGPKeyStore.addAddressToKey(@props.profile, email)
+
+  _addEmailInput: (contacts) =>
+    # when a new email is added to the list of emails
+    # flow is (click "add email") -> _addEmailClick -> (enter email) -> this
+    emails = _.pluck(contacts.to, 'email')
+    _.each(emails, @_addEmail)
+    @setState({inputEmail: false})
+
+  _addEmailClick: (event) =>
+    # create a text field in which to enter a new email to associate with a key
+    @setState({inputEmail: true})
+    # TODO focus on the new field
+    # React.findDOMNode(@refs.addNewEmail).focus()
+
+  _removeEmailClick: (event) =>
+    email = event.target.attributes['data-email'].value
+    @_removeEmail(email)
+
+  _removeEmail: (email) =>
+    PGPKeyStore.removeAddressFromKey(@props.profile, email)
```

Comment:
I think it'd be OK to merge this with the function above. It'd also be good to switch to using Actions instead of directly calling methods on the store. Usually packages create actions like this: https://github.com/nylas/N1/blob/wip/keybase/internal_packages/thread-snooze/lib/snooze-actions.js#L1, and then the store listens to the action and performs the change.